api/resource.rb

```diff
@@ -198,10 +198,17 @@ def parameters
       (@parameters || []).reject(&:exclude)
     end
 
-    def exclude_if_not_in_version(version)
+    def below_version?(version)
+      version < min_version
+    end
+
+    # Recurses through all nested properties and paramters and changes their
+    # 'exclude' instance variable if the property is at a version below the
+    # one that is passed in.
+    def exclude_if_not_in_version!(version)
       @exclude ||= version < min_version
-      @properties&.each { |p| p.exclude_if_not_in_version(version) }
-      @parameters&.each { |p| p.exclude_if_not_in_version(version) }
+      @properties&.each { |p| p.exclude_if_not_in_version!(version) }
+      @parameters&.each { |p| p.exclude_if_not_in_version!(version) }
 
       @exclude
```

Comment:
Does anywhere still require this return value? I'd expect we can remove it, and replace any occurrences that did rely on it with the `below_version?` method.